src/field_theory/krull_topology.lean

```diff
@@ -270,3 +273,73 @@ lemma krull_topology_t2 (K L : Type*) [field K] [field L] [algebra K L]
   end }
 
 end krull_t2
+
+section totally_disconnected
+
+/-- Let `X` be a topological space, and suppose that for all distinct `x,y ∈ X`, there
+  is some clopen set `U` such that `x ∈ U` and `y ∉ U`. Then `X` is totally disconnected. -/
+lemma is_totally_disconnected_of_clopen_set {X : Type*} [topological_space X]
+  (h_exists_clopen : ∀ {x y : X} (h_diff : x ≠ y), ∃ (U : set X) (h_clopen : is_clopen U),
+  x ∈ U ∧ y ∉ U) : is_totally_disconnected (set.univ : set X) :=
+begin
+  intros S _ hS,
+  by_contra,
```

Comment:
Please name the hypothesis explicitly. Also, you can merge the next line into this one by using the `by_contra'` tactic. This needs `import tactic.by_contra`.

Reply:
I named the hypothesis, but I'm not sure how to merge the lines. I played around with `by_contra'`, but I couldn't find the right usage. Could you advise me on this?

Reply:
Aah, it is indeed not possible. I pushed a golf using `push_neg`.